src/OrchardCore.Modules/Orchard.Autoroute/Drivers/AutoroutePartDisplay.cs

```diff
@@ -61,14 +65,16 @@ public override IDisplayResult Edit(AutoroutePart autoroutePart)
         public override async Task<IDisplayResult> UpdateAsync(AutoroutePart model, IUpdateModel updater)
         {
             await updater.TryUpdateModelAsync(model, Prefix, t => t.Path);
-            
+
             var httpContext = _httpContextAccessor.HttpContext;
-            
             if (httpContext != null && await _authorizationService.AuthorizeAsync(httpContext.User, Permissions.SetHomepage))
             {
                 await updater.TryUpdateModelAsync(model, Prefix, t => t.SetHomepage);
             }
 
+            var validationErrorKeyPrefix = string.IsNullOrEmpty(Prefix) ? string.Empty : Prefix + ".";
```

Comment:
*The more I look into this issue, the more I am confused. I have deleted my previous comment because some parts were incorrect.*

In content part drivers the `Prefix` property contains the name of the part e.g. `Autoroute`. 
In content field drivers the `Prefix` property contains the name of the containing part and the name of the field e.g. `SomeType.SomeNumber`

It is OK for the model binding, because in both cases we specify the last part of HTML input name in `TryUpdateModelAsync` call

`await updater.TryUpdateModelAsync(model, Prefix, t => t.Path);`

This behavior is the same even if I revert the change regarding `HtmlPrefix`.

**Validation messages**

When adding a validation error, the key must contains full name of the HTML input, so it is necessary to concatenate the prefix and the name of the property. e.g. `AutoRoute.Path` or `SomeType.SomeNumber.Value`

I think Autoroute works correctly and other field drivers should be updated to generate valid keys for validation messages. I'm not sure if there is a more elegant way than manually adding `$"{Prefix}.NameOfTheField"`. Nobody might have noticed this issue, because most of the fields doesn't have `<span asp-validation-for="Value"></span>` tag, so they can't display inline validation messages.